include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -117,41 +125,61 @@ inline void StatsdClient::gauge(const std::string& key,
     return send(key, value, "g", frequency);
 }
 
-inline void StatsdClient::timing(const std::string& key, const unsigned int ms, const float frequency) const noexcept {
+inline void StatsdClient::timing(const std::string& key, const unsigned int ms, float frequency) const noexcept {
     return send(key, ms, "ms", frequency);
 }
 
 inline void StatsdClient::send(const std::string& key,
                                const int value,
                                const std::string& type,
-                               const float frequency) const noexcept {
-    const auto isFrequencyOne = [](const float frequency) noexcept {
-        constexpr float epsilon{0.0001f};
-        return std::fabs(frequency - 1.0f) < epsilon;
-    };
-
-    // Test if one should send or not, according to the frequency rate
-    if (!isFrequencyOne(frequency)) {
-        if (frequency < static_cast<float>(std::rand()) / RAND_MAX) {
-            return;
-        }
+                               float frequency) const noexcept {
+    // Bail if we can't send anything anyway
+    if (!m_sender->initialized()) {
+        return;
+    }
+
+    // A valid frequency is: 0 <= f <= 1
+    // At 0 you never emit the stat, at 1 you always emit the stat and with anything else you roll the dice
+    frequency = std::max(std::min(frequency, 1.f), 0.f);
```

Comment:
i made `frequency` non-const so i could do this modification here, if you'd prefer it to stay `const` just let me know!

Reply:
No I guess it's better to remove the `const` keyword if we modify it internally, it makes the intent clear.